### ISSUE TITLE
deadlock in sql queries with small maxThreads

### ISSUE THREAD

**Issue by azahnen** [interactive-instruments/ldproxy#265]
With `maxThreads=1`, a single request causes the deadlock. With higher values it seems to take the same number of parallel requests that are started nearly simultaneously  to trigger it.

If possible we should of course eliminate the deadlocks altogether. If that is somehow not possible or the solution takes longer to implement, we should at least forbid values for `maxThreads` that trigger the deadlock easily, maybe everything less than `4`.

**Comment by azahnen** [comment]
The issue occurs for feature types with joins, because queries for a request are executed in parallel and then folded into a single sorted stream of rows. So the design requires the value for `maxThreads/maxConnections` to be at least `1 + number of joins`, where the feature type with the most joins counts.

Another problem is that the queries for a request are not submitted en block to the connection pool. So when `maxThreads` is set to the minimum required value described above and two parallel requests arrive, both might use some of the available connections and a deadlock occurs.

To prevent this behaviour, we will limit the thread pool that runs the feature transformation streams in the provided. The number of parallel streams that are allowed to run will be computed from `maxThreads/maxConnections` and the number of joins in the mapping.
